src/Rings/CSAs.jl

```diff
@@ -0,0 +1,231 @@
+#Structs for CentralSimpleAlgebra's and CentralSimpleAlgebraElem's
+mutable struct CentralSimpleAlgebra{T} <: NCRing
+    ideal:: Oscar.FreeAssociativeAlgebraIdeal{AbstractAlgebra.Generic.FreeAssociativeAlgebraElem{T}}
+    degree::Int
+
+    function CentralSimpleAlgebra(I::Oscar.FreeAssociativeAlgebraIdeal{AbstractAlgebra.Generic.FreeAssociativeAlgebraElem{T}}, degree:: Int) where T
+        new{T}(I,degree)
+    end
```

Comment:
Nothing about this code seems to be specific to central simple algebras, i.e. it seems to me one could make instances of this type which are not CSA. Is that intentional?

Reply:
> Did you have something specific in mind regarding this? Should I add a warning -- or a check which can be ignored by other constructors which reliably produce CSAs

No, I had nothing specific in mind yet, I was / am mostly trying to gauge what this code / what you are trying to achieve. Then based on that we can discuss and decide what's the best way to move forward: be it further improving this code, or be it replacing it with an approach based on the other existing algebra implementations.